### PR TITLE
feat(config): Use GitHub as default provider

### PR DESCRIPTION
**Pull request by BYK**
This PR switches default artifact and status providers to GitHub if the minimum required Craft version is set to 0.21.0 or above. This is *not* a breaking change for older projects.

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -299,3 +334,12 @@ export function getStatusProviderFromConfig(): BaseStatusProvider {
     }
   }
 }
+
+export function getChangelogPolicyFromConfig(): ChangelogPolicy {
+  const config = getConfiguration() || {};
```
Comment (chadwhitacre) [inline src/config.ts:339]: Why do we need the `|| {}` here but not above?

**Review by chadwhitacre — APPROVED** [review]
Seems reasonable, untested, one nit.

File: src/config.ts [hunk 2]
```diff
@@ -195,6 +201,28 @@ export function checkMinimalConfigVersion(): void {
...
+  if (!minVersionRaw) {
+    return false;
+  }
+
+  const minVersion = parseVersion(minVersionRaw);
+  if (!minVersion) {
+    throw new Error(`Cannot parse the minimal version: "${minVersionRaw}"`);
+  }
+
+  const epochVersion = parseVersion(DEFAULTS_EPOCH_VERSION);
```
Comment (jan-auer) [inline src/config.ts:216]: The `epochVersion` should probably be passed in, once you have multiple epochs. Let's say we change the default for the preReleaseCommand in one of the next versions.

File: src/config.ts [hunk 3]
```diff
@@ -299,3 +334,12 @@ export function getStatusProviderFromConfig(): BaseStatusProvider {
     }
   }
 }
+
+export function getChangelogPolicyFromConfig(): ChangelogPolicy {
+  const config = getConfiguration() || {};
+
+  return (
+    config.changelogPolicy ??
+    (isAfterEpoch() ? ChangelogPolicy.Auto : ChangelogPolicy.None)
```
Comment (jan-auer) [inline src/config.ts:343]: Let's check with SDK devs that we actually want this change. I believe we still haven't found ideal tooling to ensure changelogs are always kept up-to-date in our development flow, so a lot of repositories would rely on `simple` to error when the changelog is missing.

**Review by jan-auer — COMMENTED** [review]
Code looks good, a few comments regarding the approach below.

**Review by jan-auer — COMMENTED** [review]
Code looks good, a few comments regarding the approach below.

File: src/config.ts (hunk 1 above)
Comment (BYK) [reply src/config.ts:339]: Just copy pasted. I can do a go over for all usages.

File: src/config.ts (hunk 2 above)
Comment (BYK) [reply src/config.ts:216]: Let's revisit that when the need arises, wdyt?

File: src/config.ts (hunk 3 above)
Comment (BYK) [reply src/config.ts:343]: Fair point, I can revert this one

**Comment by BYK** [comment]
@jan-auer shall we emit a warning when we detect the old or the new default pre-epoch to urge people to upgrade?

**Review by tonyo — APPROVED** [review]

File: src/config.ts (hunk 2 above)
Comment (jan-auer) [reply src/config.ts:215]: Sure, that works for me 👍

**Review by jan-auer — APPROVED** [review]
Before merging, please update the defaults in the readme with a note to the version where this changes:

- https://github.com/getsentry/craft#status-provider
- https://github.com/getsentry/craft#artifact-provider

If I may suggest, it would be great to have a verbose message this time in the changelog notifying of the config values that are changing and what config to add in case of upgrading. Something like:

> **Breaking Change**: This version changes the defaults configuration values for `statusProvider` and `artifactProvider` to `"github"` if `minVersion` is greater or equal to `0.21.0`. If your craft configuration file does not set these providers explicitly, you can keep the old defaults by adding this to your config:
> 
> ```yaml
> minVersion: 0.21.0
> artifactProvider: zeus
> statusProvider: zeus
> ```
> 
> Support for Zeus will be dropped in a future release and we highly recommend updating your CI workflows.

File: README.md [hunk 4]
```diff
@@ -958,17 +934,33 @@ targets:
...
-    default, configurable through `releaseBranchPrefix`)
-  - Add ZEUS_HOOK_BASE as protected to CI environment
-- Add `.craft.yml` configuration file to your project
+1. Set up a workflow that builds your assets and runs your tests. Allow building
+   release branches (their names follow `release/{VERSION}` by default,
+   configurable through `releaseBranchPrefix`).
+2. Use the official `actions/upload-artifact@v2` action to upload your assets.
+   Here is an example config (step) of an archive job:
+
+    ```yaml
```
Comment (jan-auer) [inline README.md:943]: Indentation is one off, does it copy/paste fine from the rendering?

**Review by jan-auer — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (BYK) [reply README.md:943]: Possible. Will fix.

**Pull request merged**